File: media/js/firefox/new/basic/download.js
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

(function() {
    'use strict';

    var client = window.Mozilla.Client;
    var otherPlatformsLink = document.querySelector('.js-platform-modal-button');
    var otherPlatformsContent = document.querySelector('.other-platforms-content');

    var initOtherPlatformsModal = function() {
        // show the modal cta button
        otherPlatformsLink.classList.remove('hidden');

        otherPlatformsLink.addEventListener('click', function(e) {
            e.preventDefault();

            Mzp.Modal.createModal(this, otherPlatformsContent, {
                title: otherPlatformsLink.textContent,
                className: 'other-platforms-modal'
            });

            window.dataLayer.push({
                'event': 'in-page-interaction',
                'eAction': 'link click',
                'eLabel': 'Download Firefox for another platform'
            });
        }, false);
    };

    /**
     * Enable modal to optionally download Firefox for other platforms.
     * Don't show the modal for iOS or Android.
     */
    if (otherPlatformsLink && client.isDesktop) {
        initOtherPlatformsModal();
    }

})();

File: tests/pages/firefox/download/download.py
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at https://mozilla.org/MPL/2.0/.

from selenium.webdriver.common.by import By

from pages.base import BasePage


class DownloadPage(BasePage):
    _URL_TEMPLATE = "/{locale}/firefox/download/"

    _download_button_locator = (By.CSS_SELECTOR, "#download-button-thanks > .download-link")

    @property
    def is_download_button_displayed(self):
        return self.is_element_displayed(*self._download_button_locator)
